pycroft/lib/finance/retransfer.py

```diff
@@ -46,3 +51,27 @@ def generate_activities_return_sepaxml(activities: list[BankAccountActivity]) ->
         sepa.add_payment(payment)
 
     return sepa.export()
+
+
+def attribute_activities_as_returned(
+    session: Session, activities: list[BankAccountActivity], author: User
+) -> None:
+    for activity in activities:
+        debit_account = config.non_attributable_transactions_account
+        credit_account = activity.bank_account.account
+
+        transaction = simple_transaction(
+            description=activity.reference,
+            debit_account=debit_account,
+            credit_account=credit_account,
+            amount=activity.amount,
+            author=author,
+            valid_on=activity.valid_on,
+            confirmed=False,
+        )
+        activity.split = next(
+            split for split in transaction.splits if split.account_id == credit_account.id
+        )
+        session.add(activity)
+
+    session.commit()
```

Comment:
Please do not commit the session in the `lib`.
While it is true that one should commit often, we cannot judge inside such a function what the lifecycle of the transaction should look like.

For instance, in the following finance endpoint, multiple `lib` functions will be called together:
https://github.com/agdsn/pycroft/blob/865ebcf6bfc5a7a3b02966bf64e00c2342b42717/web/blueprints/finance/__init__.py#L509-L525

You notice that these are lots of operations and only at the end will we judge the transaction to be complete.
Sure, this should be its own lib function, but then the same argument would apply to `end_payment_in_default_memberships`, which cannot know how it will be used.